sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -327,6 +327,223 @@ public CosmosClientBuilder contentResponseOnWriteEnabled(boolean contentResponse
+     * Sets the DIRECT connection configuration to be used.
+     *
+     * @param directConnectionConfig DIRECT connection configuration
+     * @return current CosmosClientBuilder
+     */
+    public CosmosClientBuilder directMode(DirectConnectionConfig directConnectionConfig) {
+        this.directConnectionConfig = directConnectionConfig;
+        return this;
+    }
+
+    /**
+     * sets the value of the user-agent suffix.
+     *
+     * @param userAgentSuffix The value to be appended to the user-agent header, this is
+     * used for monitoring purposes.
+     *
+     * @return current CosmosClientBuilder
+     */
+    public CosmosClientBuilder userAgentSuffix(String userAgentSuffix) {
+        this.userAgentSuffix = userAgentSuffix;
+        return this;
+    }
+
+    /**
+     * Sets the retry policy options associated with the DocumentClient instance.
+     * <p>
+     * Properties in the RetryOptions class allow application to customize the built-in
+     * retry policies. This property is optional. When it's not set, the SDK uses the
+     * default values for configuring the retry policies.  See RetryOptions class for
+     * more details.
+     *
+     * @param throttlingRetryOptions the RetryOptions instance.
+     * @return current CosmosClientBuilder
+     * @throws IllegalArgumentException thrown if an error occurs
+     */
+    public CosmosClientBuilder throttlingRetryOptions(ThrottlingRetryOptions throttlingRetryOptions) {
+        this.throttlingRetryOptions = throttlingRetryOptions;
+        return this;
+    }
+
+    /**
+     * Sets the preferred regions for geo-replicated database accounts. For example,
+     * "East US" as the preferred region.
+     * <p>
+     * When EnableEndpointDiscovery is true and PreferredRegions is non-empty,
+     * the SDK will prefer to use the regions in the collection in the order
+     * they are specified to perform operations.
+     * <p>
+     * If EnableEndpointDiscovery is set to false, this property is ignored.
+     *
+     * @param preferredRegions the list of preferred regions.
+     * @return current CosmosClientBuilder
+     */
+    public CosmosClientBuilder preferredRegions(List<String> preferredRegions) {
+        this.preferredRegions = preferredRegions;
+        return this;
+    }
+
+    /**
+     * Sets the flag to enable endpoint discovery for geo-replicated database accounts.
+     * <p>
+     * When EnableEndpointDiscovery is true, the SDK will automatically discover the
+     * current write and read regions to ensure requests are sent to the correct region
+     * based on the capability of the region and the user's preference.
+     * <p>
+     * The default value for this property is true indicating endpoint discovery is enabled.
+     *
+     * @param endpointDiscoveryEnabled true if EndpointDiscovery is enabled.
+     * @return current CosmosClientBuilder
+     */
+    public CosmosClientBuilder endpointDiscoverEnabled(boolean endpointDiscoveryEnabled) {
```

Comment:
endpointDiscoveryEnabled

Reply:
Good catch @milismsft - You saved us V5 🤣